### ISSUE TITLE
AWS EC2 API doesn't allow encrypted parameter when specifying a snapshot id

### ISSUE THREAD

**Issue by vanga** [ansible/ansible-modules-core#1693]
#### Issue Type:

Bug Report
#### Ansible version:

ansible 2.0.0 (devel de52cfb5d7)
But the ec2.py is updated with this #1534. So Yes, I am able to create encrypted volumes.
#### Ansible Configuration:

Default
##### Environment:

ArchLinux on EC2
##### Summary:

If we attach a volume with a snapshot id specified, AWS doesn't give an option to enable/disable encryption. I think AWS just decides the value for `encrypted` parameter based on whether snapshot is encrypted or not.
![screenshot from 2015-07-09 19 45 04](https://cloud.githubusercontent.com/assets/5126396/8597475/36c4de00-2673-11e5-9428-8f2b3ed5e2ec.png)
##### Steps To Reproduce:

What I did exactly is, created an AMI with an attached EBS, tried to launch new instance with this AMI by passing a volume dict containing the snapshot ID

tasks/main.yml

```
    user_data: "{{ user_data | default(omit) }}"
    volumes: "{{volumes | default (omit)}}"
    wait: yes
```

playbook

```
 roles:
    - role: launch
      launch_for: "elasticsearch"
      volumes:
      - device_name: /dev/sdb
        device_type: gp2
        volume_size: 20
        delete_on_termination: false
```

I also tried removing the change made in #1534 , It worked, I was able to launch a new instance.
So, I guess we have to write a conditional to not pass encrypted parameter when sending a snapshot id

I think this issue is similar https://github.com/hashicorp/terraform/issues/1862
##### Actual Results:

Error

```
failed: [localhost -> 127.0.0.1] => {"failed": true}
msg: Instance creation failed => InvalidParameter: Parameter encrypted is invalid. You cannot specify the encrypted flag if specifying a snapshot id in a block device mapping.

FATAL: all hosts have already failed -- aborting
```

**Comment by vanga** [comment]
Ok, I see that a change is pushed recently https://github.com/ansible/ansible-modules-core/commit/b4911a47d1589693791bdf40ed979e239d69d6d2

It works for me with the above change, but it doesn't work if some one passes `encrypted` parameter unknowingly

I think correct way of doing it would be to throw an error if some one tries to set encryption, I added a conditional to check the above case https://github.com/ansible/ansible-modules-core/pull/1694

**Comment by bcoca** [comment]
closed via #1694 merge
